db/updateItem.go

```diff
@@ -39,8 +41,46 @@ func (d *dynamoDB) UpdateItem(listID string, itemID string, newName string) (*da
 		if e.Code() == dynamodb.ErrCodeConditionalCheckFailedException {
 			return nil, ErrorNotFound
 		}
+		if e.Code() == "ValidationException" { // https://github.com/aws/aws-sdk-go/issues/3140
+			return nil, ErrorBadRequest
+		}
+		return nil, err
 	default:
 		return nil, err
 	}
-	return item, nil
+
+	item := new(data.Item)
+	err = dynamodbattribute.UnmarshalMap(output.Attributes, &item)
+	return item, err
+}
+
+func getUpdateFields(newName string, isCompleted *bool) (map[string]*dynamodb.AttributeValue, *string, map[string]*string) {
+	fields := map[string]*dynamodb.AttributeValue{}
+	names := map[string]*string{}
+	var updateExpression *string
+
+	if isCompleted != nil {
+		fields[":c"] = &dynamodb.AttributeValue{BOOL: isCompleted}
+		updateExpression = appendUpdateExpression(updateExpression, "IsCompleted = :c")
+	}
+
+	if newName != "" {
```

Comment:
We would need one `if` per field that could be updated, this might not scale if we had loads but I can't imagine us having more than a couple more